flecsi/topology/common/utility_types.h

```diff
@@ -94,5 +94,79 @@ using id_vector_map_t =
 
 using index_vector_t = std::vector<size_t>;
 
+//-----------------------------------------------------------------//
+//! \class entity_base_u types.h
+//! \brief entity_base_u defines a base class that stores the raw info that
+//! a topology needs, i.e: id and rank data
+//!
+//! \tparam N The number of domains.
+//-----------------------------------------------------------------//
+
+class entity_base_
```

Comment:
There are `entity_base_` and `entity_base_u` classes already defined in flecsi/topology/types.h.  Can you use those instead of redefining them here?  I added the types.h definitions relatively recently, so it's possible that those definitions weren't present when you started your work, but they are available now.

Reply:
I looked a little closer - my comment above is based on master, but you are merging to feature/refactor, which was apparently branched off of master before I added my definitions there.  To further confuse matters, it looks like the old flecsi/topology/types.h got renamed to flecsi/topology/common/utility_types.h.

Could you please check and verify that the changes you are adding here are consistent with the ones I made in master?  That way we can avoid a merge headache later.

Reply:
@cferenba I made sure of the consistency of the two versions.